### PR TITLE
Action to publish docker image

### PR DESCRIPTION
**Pull request by DCjanus**
Resolve: https://github.com/bnjbvr/cargo-machete/issues/93

I have already tested the code of this Pull Request in my own namespace. For details, please see this URL: https://github.com/DCjanus/cargo-machete/pkgs/container/cargo-machete

Since CI is a module that is difficult to debug locally, I have submitted many fragmented commits in this PR, hoping that all my commits will be **squashed** upon merging.

### CODE REVIEW

**Comment by DCjanus** [comment]
Since CI is a module that is difficult to debug locally, I have submitted many fragmented commits in this PR, hoping that all my commits will be squashed upon merging.

**Comment by bnjbvr** [comment]
Hi! Thanks for the contribution, out of curiosity why do you need a Docker image? I think the linked issue has been closed because it was deemed possible to use another step for Github CI workflows, and I'm not sure to see the benefits of using a Docker instance locally — but I might be missing something!

**Comment by DCjanus** [comment]
For some enterprises with strict network isolation, it may be difficult to directly download a binary from GitHub. However, it is relatively easy to copy a binary from another public docker image because the enterprise can build its own mirror internally.

**Comment by bnjbvr** [comment]
Ok, makes sense! I've also received good feedback for a docker image for cargo-machete on the fediverse, so I'm happy taking this change.

in terms of process: what does it entail? IIUC, i would need to set up a new Docker repository at bnjbvr/cargo-machete too. Are there other steps that would be required? (like adding the github bot as a collaborator to the Docker repo, maybe? if so, how does one do that?)

There are a few TODOs in your PR, do you plan to implement them or can/should they be done later, when actually required?

**Comment by DCjanus** [comment]
If you are only publishing the image to ghcr.io, you don't need to take any action except merging this PR.

I prefer to implement the TODOs in the future in order to avoid introducing too many changes in one PR.

**Comment by DCjanus** [comment]
> I've also received good feedback for a docker image for cargo-machete on the fediverse

For people who want to gain more idea about this, I'd like to put the link about this feedback here.

https://tutut.delire.party/@bnjbvr/112394931314428509

File: README.md [hunk 1]
```diff
@@ -78,6 +79,17 @@ flag, which will call `cargo metadata --all-features` to find final dependency
 names, more accurate dependencies per build type, etc. ⚠ This may modify the
 `Cargo.lock` files in your projects.
 
+## Cargo Machete Image
```
Comment (bnjbvr) [inline README.md:82]:
```suggestion
## Docker Image
```

File: README.md [hunk 2]
```diff
@@ -78,6 +79,17 @@ flag, which will call `cargo metadata --all-features` to find final dependency
...
+
+A docker image for cargo machete.
+
+### Example usage
+
+The command given by,
```
Comment (bnjbvr) [inline README.md:88]:
```suggestion
For instance, run `cargo-machete` in the `$(pwd)` directory using:

```

**Review by bnjbvr — APPROVED** [review]
Thanks!

**Pull request merged**